src/main/Application.h

```diff
@@ -164,6 +164,16 @@ class Application
         APP_NUM_STATE
     };
 
+    // Types of threads that may be running
+    enum class ThreadType
+    {
+        MAIN,
+        WORKER,
+        EVICTION,
+        OVERLAY,
+        LEDGER_CLOSE
```

Comment:
In conversation with @SirTyson a few minutes ago we were trying to figure out terminology that works best for differentiating the state-bits being touched by the "close thread" vs. the lagging LCL, and we came to the conclusion that a good naming scheme would be to keep LCL as the lagging state (it's already used everywhere) and just change the _new_ concept of the database-and-buckets (which can get ahead of LCL) to be the last _applied_ ledger. So "close" or "closed" terminology continues to mean "data fully published back to the main thread state" and "apply" is reserved for the actions of this extra thread. And the thread doing it is therefore the "apply" thread, not the "close" thread. (And I will follow up with a few method-renames inside `LedgerManagerImpl` to reflect this also). 
```suggestion
        APPLY
```